### PR TITLE
Restructure Preprocessing Example Analysis

### PR DESCRIPTION
**Pull request by roshankern**
In this PR we restructure the example analysis for preprocessing to have GitHub-contained example data. 

This follows the precedents for example analysis set in #8 and #9.

Notes:
- For this PR review I have removed the example files. Once the PR review is done I will add them back. They prevent a reviewer from seeing differences.
- Some of the input ND2 files were too large to include in this repo, so we will have to come up with a different method to make these accessible to outside users.

@mat10d This is ready for review, let me know if you have any questions!

### CODE REVIEW

File: example_analysis/0_preprocessing/0_preprocessing.smk
Comment (mat10d) [inline example_analysis/0_preprocessing/0_preprocessing.smk:1]: `# Final output files
rule all:
    input:
        expand(
            f"{METADATA_DIR}/10X_c{{cycle}}-SBS-{{cycle}}_{{well}}.metadata.pkl",
            well=WELLS,
            cycle=SBS_CYCLES,
        ),
        expand(
            f"{METADATA_DIR}/20X_{{well}}.metadata.pkl",
            well=WELLS,
        ),
        expand(
            f"{SBS_TIF_DIR}/10X_c{{cycle}}-SBS-{{cycle}}_{{well}}_Tile-{{tile}}.sbs.tif",
            well=WELLS,
            cycle=SBS_CYCLES,
            tile=SBS_TILES,
        ),
        expand(
            f"{PH_TIF_DIR}/20X_{{well}}_Tile-{{tile}}.phenotype.tif",
            well=WELLS,
            tile=PH_TILES,
        ),
        expand(
            f"{IC_DIR}/10X_c{{cycle}}-SBS-{{cycle}}_{{well}}.sbs.illumination_correction.tif",
            well=WELLS,
            cycle=SBS_CYCLES,
            tile=SBS_TILES,
        ),
        expand(
            f"{IC_DIR}/20X_{{well}}.phenotype.illumination_correction.tif",
            well=WELLS,
            tile=PH_TILES,
        ),`

This is a really good way of reshaping our all rules. Makes it much more readable

File: ops/preprocessing_smk.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import inspect
 import functools
 import os
-import warnings
+import re
+
 import pandas as pd
 import numpy as np
-import re
 from nd2reader import ND2Reader
```
Comment (mat10d) [inline ops/preprocessing_smk.py:8]: The one change we want to make here is switching from nd2Reader to nd2. The tile based pre-processing is way easier than the well based one, so you may want to weigh up if you want to make that switch when you rewrite in the new repo: https://www.talleylambert.com/nd2/API/nd2/. Talley Lambert is one of the most well known computational microscopy people around, and we can be sure that this package will be maintained long term

Comment (roshankern) [reply ops/preprocessing_smk.py:8]: Thanks for pointing this out! I like the idea of switching from nd2Reader to nd2 when rewriting preprocessing in the new repo.

**Comment by roshankern** [comment]
<p>Yep the goal of this does make sense and I remember seeing in the previous example file that the cell patterns lined up for part of the images (these are the cells that have overlapping data for SBS and PH image iirc). </p><p> </p><blockquote>I am now thinking that maybe this isn't something we need to do here, and instead could be done in the merge/hash step. However, just showing a table of all of the ph and sbs tiles doesn't really make much sense. The goal is to show ones that will align between 10x and 20x, and then essentially display a set of matching ones so that you can see the pattern of cells aligning between the two tiles you selected.</blockquote><p> </p><p>Would you recommend we just leave the grid views out of these new examples that don't have that much data? Especially if we want to move this to the merge/hash step? If so I can just remove it from this example analysis eval notebook.</p>

---

 <a href='https://app.reviewnb.com/cheeseman-lab/OpticalPooledScreens/pull/12/discussion/#comment-2405432180'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by mat10d** [comment]
<p>Yeah, let's just exclude it for now.</p>

---

 <a href='https://app.reviewnb.com/cheeseman-lab/OpticalPooledScreens/pull/12/discussion/#comment-2405432180'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by roshankern** [comment]
<p>Done in <a href="https://github.com/cheeseman-lab/OpticalPooledScreens/pull/12/commits/bef2bd2a5c5a0829458aff49dd878f6f6458a513" target="_blank">https://github.com/cheeseman-lab/OpticalPooledScreens/pull/12/commits/bef2bd2a5c5a0829458aff49dd878f6f6458a513</a></p>

---

 <a href='https://app.reviewnb.com/cheeseman-lab/OpticalPooledScreens/pull/12/discussion/#comment-2405432180'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by roshankern** [comment]
<p>Removed these examples in <a href="https://github.com/cheeseman-lab/OpticalPooledScreens/pull/12/commits/bef2bd2a5c5a0829458aff49dd878f6f6458a513" target="_blank">https://github.com/cheeseman-lab/OpticalPooledScreens/pull/12/commits/bef2bd2a5c5a0829458aff49dd878f6f6458a513</a></p>

---

 <a href='https://app.reviewnb.com/cheeseman-lab/OpticalPooledScreens/pull/12/discussion/#comment-2405432213'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Pull request merged**